app/controllers/claims_controller.rb

```diff
@@ -67,8 +67,7 @@ def on_school_flow?
   end
 
   def next_slug
-    current_slug_index = current_claim.page_sequence.index(params[:slug])
-    current_claim.page_sequence[current_slug_index + 1]
+    PageSequence.new(current_claim, params[:slug]).next_slug
```

Comment:
This feels like we crate a new PageSequence everytime we call `next_slug`, I think I would've created a single PageSequence instance when the claims controller is inatilized? What's the thinking behind this approach? Does Rails create a new controller for each request so `next_slug` is only ever called once?

Reply:
Yep, each request instantiates a new instance of the controller, and so because we only call `next_slug` once, this only gets instantiated once. We may want to memoize this and/or extract it to a helper method at some point, but there is no reason to do so now.